platform/viewer/src/index.js

```diff
@@ -15,19 +15,21 @@ import ReactDOM from 'react-dom';
  * pluginImports.js imports all of the modes and extensions and adds them
  * to the window for processing.
  */
-import './pluginImports.js';
+import loadDynamicImports from './pluginImports.js';
 
-/**
- * Combine our appConfiguration with installed extensions and modes.
- * In the future appConfiguration may contain modes added at runtime.
- *  */
-const appProps = {
-  config: window ? window.config : {},
-  defaultExtensions: window.extensions,
-  defaultModes: window.modes,
-};
+loadDynamicImports().then(() => {
```

Comment:
This code change here is the ONLY active running code that is changed in this PR, and it simply calls a function to load any dynamic imports.   How that function does the load isn't currently defined, but it would be possible to merge some of the code from OHIFModesAndExtensions into core OHIF to make this possible.